### PR TITLE
Drizzle to Prisma Migration

### PR DESCRIPTION
**Pull request by shiyaamsunder**
**This is a breaking change**

This PR migrates the previously set up Drizzle ORM to Prisma. All the drizzle related files are kept as is incase something goes wrong. I have tested almost all the migrated queries and the functionalities locally. So far looks good to me.

---
### Prisma 
image

Prisma usually puts the schema and the migration files into the `prisma` folder. 

---
### UI Changes

Redone the `/create` page UI to match our application theme. Removed the `marks` like link, unset link from TipTap which were not working for now.  This is done as part of issue #28. 

Regarding TipTap Editor, all the code was in normal JS and the styling was done in plain CSS. So I migrated it to typescript and tailwind css. (note to @naveenr03 )

image

Changes were made to the show the correct Topic name and Thread name in respective places.

image
image

---

### DB changes

There is no relational changes, instead only name changes and attribute changes for brevity purposes. But as I mentioned before. `role_ids` in the `User` table should be decomposed in to a separate joining table. This needs to be discussed further.

**Drizzle DB relations**
old

**Prisma DB relations**
new

---

### File Structure changes

Moved these files to `components` folder

- moved `src/app/home/Section.tsx -> src/components/section.tsx`
- moved `src/app/home/TipTapInput.tsx -> src/components/create-thread.tsx`
- moved `src/components/TipTap.jsx -> src/components/tiptap/tiptap-core.tsx` 
- renamed `TipTapInput to TipTapCore`
- removed `TipTap.css`

image

image

---

Stepping forward we don't need to worry abt the ORM and start focusing the main features of the applications. :)

### CODE REVIEW

File: prisma/migrations/20240727152332_init/migration.sql
Comment (Gukkey) [inline prisma/migrations/20240727152332_init/migration.sql:1]: Where are triggers for updating updated_at columns?

Comment (shiyaamsunder) [reply prisma/migrations/20240727152332_init/migration.sql:1]: Did you send it?

Comment (Gukkey) [reply prisma/migrations/20240727152332_init/migration.sql:1]: https://github.com/Gukkey/project-forum/blob/1711c5981013a1c7429056090617bb1092517a20/drizzle/0004_quiet_winter_soldier.sql#L11-L47

I think we did talk about that during a discussion before starting/during the migration. Linked the triggers btw. If you don't mind I can directly commit them, I don't know how to suggest those changes like people do.

Comment (Gukkey) [reply prisma/migrations/20240727152332_init/migration.sql:1]: Made a commit for this. This is resolved now.

**Review by Gukkey — APPROVED** [review]

**Pull request merged**